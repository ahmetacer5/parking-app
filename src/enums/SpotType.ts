export enum SpotType {
    Compact = 'Compact',
    Large = 'Large',
    Handicapped = 'Handicapped',
    Motorcycle = 'Motorcycle',
}
