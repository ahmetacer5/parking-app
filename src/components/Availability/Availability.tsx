import { observer } from 'mobx-react-lite';
import React, { FC } from 'react';
import { Button, Grid } from 'theme-ui';

import { CarIcon, MotorcycleIcon, VanIcon, WheelchairIcon } from '../../assets/icons';
import { SpotType } from '../../enums';
import { AvailabilityNumbers } from '../../types/AvailabilityNumbers';

interface DisplayProps {
    spotType: SpotType;
    value?: number;
    onChange?: (spotType: SpotType, decrease?: boolean) => void;
}
const Display: FC<DisplayProps> = ({ spotType, value = 0, onChange }) => {
    const onIncreaseHandler = () => onChange && onChange(spotType);
    const onDecreaseHandler = () => onChange && onChange(spotType, true);
    return (
        <Grid color={value > 0 ? 'available' : 'unavailable'}>
            {onChange && (
                <Button variant="tiny" onClick={onIncreaseHandler}>
                    +
                </Button>
            )}
            {spotType === SpotType.Compact && <CarIcon />}
            {spotType === SpotType.Large && <VanIcon />}
            {spotType === SpotType.Handicapped && <WheelchairIcon />}
            {spotType === SpotType.Motorcycle && <MotorcycleIcon />}
            <span data-testid={`${spotType}-value`}>{value}</span>
            {onChange && (
                <Button variant="tiny" onClick={onDecreaseHandler}>
                    -
                </Button>
            )}
        </Grid>
    );
};

export interface AvailabilityProps extends Pick<DisplayProps, 'onChange'> {
    numbers: AvailabilityNumbers;
}

export const Availability: FC<AvailabilityProps> = observer(({ numbers, onChange }) => {
    return (
        <Grid variant="styles.availabilityCard">
            {Object.keys(numbers).map((type) => (
                <Display key={type} spotType={type as SpotType} onChange={onChange} value={numbers[type as SpotType]} />
            ))}
        </Grid>
    );
});
