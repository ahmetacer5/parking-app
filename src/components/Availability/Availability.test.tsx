import '@testing-library/jest-dom/extend-expect';

import { render } from '@testing-library/react';
import React from 'react';

import { SpotType } from '../../enums';
import { Availability, AvailabilityProps } from './Availability';

const defaultProps: AvailabilityProps = {
    numbers: { [SpotType.Motorcycle]: 10, [SpotType.Compact]: 5 },
};

describe('Availability', () => {
    test('Should match with the snapshot', () => {
        const { container } = render(<Availability {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });
    test('Renders numbers correctly', () => {
        const { queryByTestId } = render(<Availability {...defaultProps} />);
        expect(queryByTestId(`Motorcycle-value`)).toHaveTextContent('10');
        expect(queryByTestId(`Compact-value`)).toHaveTextContent('5');
    });
});
