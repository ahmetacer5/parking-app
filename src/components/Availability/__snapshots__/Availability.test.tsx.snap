// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Availability Should match with the snapshot 1`] = `
<div>
  <div
    class="css-1dlftso"
  >
    <div
      class="css-wphdyd"
    >
      <svg>
        motorcycle.svg
      </svg>
      <span
        data-testid="Motorcycle-value"
      >
        10
      </span>
    </div>
    <div
      class="css-wphdyd"
    >
      <svg>
        car.svg
      </svg>
      <span
        data-testid="Compact-value"
      >
        5
      </span>
    </div>
  </div>
</div>
`;
